### PR TITLE
Fix - SAML - use username (optional) in attributes mapper

### PR DESCRIPTION
**Pull request by Belgor**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

In User/Authentication Settings - SAML SSO Settings is ignored attribute mapping for **Username (optional)**
Condition in code is designed way, that Username cannot be never used since Email is required attribute.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Set up SAML with Username attribute mapping - see docs [here](https://docs.mautic.org/en/authentication) for instructions
2. Log in with new account using SSO
3. Username is email

#### Steps to test this PR:
1. Load up [this PR](https://m3.mautibox.com)
2. Set up SAML with Username attribute mapping
3. Log in with new account using SSO
4. Username is username

### CODE REVIEW

**Comment by mabumusa1** [comment]
@Belgor what do you need help with?

**Comment by Belgor** [comment]
> @Belgor what do you need help with?

Hello @mabumusa1 
I just wanted to invoke a pull request of fix for SAML feature we use.
In our implementation we noticed that SAML do not use "username" so i made a fix for Mautic v 3
I am not a member with enough rights so i am waiting for some code review and approval.

Is everything done right from my side?

**Review by mabumusa1 — DISMISSED** [review]
I tested this case using GSuite, it worked as expected

**Comment by mabumusa1** [comment]
> > @Belgor what do you need help with?
> 
> Hello @mabumusa1
> I just wanted to invoke a pull request of fix for SAML feature we use.
> In our implementation we noticed that SAML do not use "username" so i made a fix for Mautic v 3
> I am not a member with enough rights so i am waiting for some code review and approval.
> 
> Is everything done right from my side?

I do not have enough rights, but I wrote my review.

I think @RCheesley can help us with this

**Comment by RCheesley** [comment]
Hi folks, need one more test to be made! Also we don't have a code coverage report - sometimes rebasing the PR is enough to trigger it!  Bug fixes go into the 3.0 branch now. fyi :)

**Comment by RCheesley** [comment]
I added a link in the description to the docs on setting up SAML - https://docs.mautic.org/en/authentication. I wasn't able to test this as I don't have an SSL localhost setup but will try to find someone to confirm!

**Comment by mabumusa1** [comment]
@Rcheesley I used this in production it worked

**Comment by RCheesley** [comment]
> @RCheesley I used this in production it worked

Yep, we just need one more tester to confirm then we can get it merged :)

**Review by dennisameling — APPROVED** [review]
Code looks good 👍 if an email is set, it will be used as username. If a username is set and it's not empty, that will be used as the username. Thanks @Belgor!

**Pull request merged**
